rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -547,6 +568,68 @@ def _decode_curve_trades(self, context: DecoderContext) -> DecodingOutput:
+        received. We need to match against transfers manually because they are not
+        made from the user address but an intermediate contract
+        """
+        if (pool_addresses := self.pools.get(context.tx_log.address)) is None:
+            log.error(
+                f'Curve pool for {self.evm_inquirer.chain_name} {context.tx_log.address} '
+                f'not present in cache at {context.transaction.tx_hash.hex()}. Skipping',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        provider = hex_or_bytes_to_address(context.tx_log.topics[1])
+        deposited_amounts = [
+            deposited_amount
+            for deposit_amount in [context.tx_log.data[i:i + 32] for i in range(len(pool_addresses))]  # noqa: E501
+            if (deposited_amount := hex_or_bytes_to_int(deposit_amount)) != 0
+        ]
+        pool_assets = [
+            Asset(evm_address_to_identifier(
+                address=address,
+                chain_id=self.evm_inquirer.chain_id,
+                token_type=EvmTokenKind.ERC20,
+            )) if address != ETH_SPECIAL_ADDRESS else A_ETH for address in pool_addresses
+        ]
+
+        gauge_tokens = None
+        for event in context.decoded_events:
+            if event.asset not in pool_assets:
+                continue
+
+            # The context.tx_log is a deposit event from curve. We deposit an underlying token
+            # of the pool (for example EURE) and we get an amount of the pool token. The problem
+            # is that the event log doesn't contain the amount of the LP received and it is needed
+            # later when decoding the gauge events since the amounts that we compare with are
+            # different. The next for loop iterates over the logs to find the transfer that follows
+            # the current log entry and extract from there the amount of LP tokens obtained after
+            # the deposit.
+            for tx_log in context.all_logs:
+                if (
+                    tx_log.log_index > context.tx_log.log_index and
+                    tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                    hex_or_bytes_to_address(tx_log.topics[1]) == provider
+                ):
+                    gauge_tokens = hex_or_bytes_to_int(tx_log.data[0:32])
+                    break
+
+            # add the amount of gauge_tokens obtained, to the event's extra_data
+            for deposit_amount in deposited_amounts:
+                if event.balance.amount == asset_normalized_value(
+                    amount=deposit_amount,
+                    asset=event.asset.resolve_to_crypto_asset(),
+                ):
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.counterparty = CPT_CURVE
+                    event.extra_data = {'gauge_tokens': gauge_tokens}
+                    break
```

Comment:
again here. If you don't find the event to decorate with gauge_tokens shouldn't you log something?